django/test/runner.py

```diff
@@ -682,14 +682,18 @@ def suite_result(self, suite, result, **kwargs):
         return len(result.failures) + len(result.errors)
 
     def _get_databases(self, suite):
-        databases = set()
+        databases = {}
         for test in suite:
             if isinstance(test, unittest.TestCase):
                 test_databases = getattr(test, 'databases', None)
                 if test_databases == '__all__':
-                    return set(connections)
+                    test_databases = connections
                 if test_databases:
-                    databases.update(test_databases)
+                    serialized_rollback = getattr(test, 'serialized_rollback', False)
+                    databases.update(
+                        (alias, serialized_rollback or databases.get(alias, False))
+                        for alias in test_databases
+                    )
             else:
                 databases.update(self._get_databases(test))
```

Comment:
FYI, I could be wrong, but it looks like this `update()` line could wind up overwriting the special `update()` precedence logic you include in the other branch. A possible fix for this would be to change `_get_databases()` to accept a required dict of databases and initialize the dict outside the first call. That way the update will only be happening in one place, and to the same dict. A second option would be to have these recursive calls build a list of pairs, and combine them to a dict only at the very end, in an outer call.

Reply:
@cjerdonek you are right. I guess passing `databases` in calls and defaulting to a new `dict` if not provided would also work.

Reply:
Yes, you could do that, too. It's a little confusing because `_get_databases()` functions more like "update" (the provided argument) when a dict is provided. The return value isn't really used in that case.

Reply:
This was actually dead code from the start.